_posts/2022-09-07-quick-tips-to-speed-up-ci.markdown

```diff
@@ -0,0 +1,145 @@
+---
+layout: epic
+title: "Parallelizing Jest and Cypress.io Tests on CircleCI"
+subtitle: A few simple tips on how to speed up your CI
+date: 2022-09-07
+categories: [CircleCI, CI, Tests, Jest, Cypress]
+author: [chris]
+---
+
+At Artsy, exploring ways to improve the developer experience is part of our
+makeup. Whether it's implementing [hot-swapping][hot-swap] for Express.js or
+integrating the Rust-based [SWC compiler][swc] into our front-end build
+pipeline, we're always trying to reduce the amount of time it takes for a code
+cycle to take place. CI is no exception. When a developer opens a PR, we want to
+ensure they get timely feedback. Have their unit tests passed? Does the app
+build correctly? And how about smoke tests? Each of these jobs are complex
+processes that take time, and the more one can parallelize said tasks the less
+waiting devs will need to do. Scaled out to a whole engineering org, minor
+improvements to CI can be radical.
+
+In this regard, two things came across our radar recently that we'd like to
+share: sharding via Jest, and a (free) way to parallelize Cypress.io integration
+tests via CircleCI's `split` command.
+
+<!-- more -->
+
+## Sharding in Jest
+
+"What is sharding?" Good question! In short, it means "a small part of a whole".
+The database community has employed sharding techniques for decades, where a
+large database is split up into smaller, more manageable chunks, usually to
+improve performance at scale. The same idea can be applied to any process or
+task involving a lot of data, including tests.
+
+Think about it like this. Imagine an app that has thousands of tests. One can
+open up their terminal and run `yarn test` and execute all of the tests at once
+in a single process, or one can open two terminal tabs and run
+`yarn test src/utils` and `yarn test src/routes`, and have both processes
+allocate a pool of memory to complete each (smaller) subset of tasks. Because
+each process has its own memory pool the performance characteristics are
+generally better, and thus speed up the overall time required to run our tests.
+To run each of these commands scoped to a particular folder is easy enough, but
+in a CI environment this is somewhat cumbersome; we'd need to define two new
+jobs and then the conditions in which they run, increasing the scope and
+complexity of our configuration file.
+
+This is where Jest's new sharding feature comes into play, which taps nicely
```

Comment:
Wow, I missed this feature!

**suggestion (non-blocking):** maybe link above to their [announcement](https://jestjs.io/blog/2022/04/25/jest-28#sharding-of-test-run) or the api docs?